hack/e2e/kops.sh

```diff
@@ -69,3 +78,36 @@ function kops_delete_cluster() {
   loudecho "Deleting cluster ${CLUSTER_NAME}"
   ${KOPS_BIN} delete cluster --name "${CLUSTER_NAME}" --state "${KOPS_STATE_FILE}" --yes
 }
+
+# TODO switch this to python, all this hacking with jq stinks!
+function kops_patch_cluster_file() {
+  CLUSTER_FILE=${1}
+  KOPS_PATCH_FILE=${2}
+
+  loudecho "Patching cluster $CLUSTER_NAME with $KOPS_PATCH_FILE"
+
+  # Temporary intermediate files for patching
+  CLUSTER_FILE_0=$CLUSTER_FILE.0
```

Comment:
What does `$CLUSTER_FILE.0` do here?

Reply:
I can't open the file for writing via bash redirect and read it at the same time  to "patch" it hence this convoluted 'intermediate files' idea.